<?xml version="1.0"?>

<launch>
  <!-- Bluerov robot transforms -->
  <include file="$(find bluerov)/launch/description.launch" />

  <!-- Mavros -->
  <include file="$(find bluerov_bringup)/launch/apm.launch" />

  <node pkg="nodelet" type="nodelet" name="bluerov_nodelet_manager" args="manager" output="screen" respawn="true">
      <param name="num_worker_threads" value="3" />
  </node>

  <include file="$(find minimal_bluerov_control_experiment)/stag.launch"/>

  <include file="$(find spinnaker_sdk_camera_driver)/launch/acquisition.launch">
    <arg name="nodelet_manager_name" value="bluerov_nodelet_manager"/> 
    <arg name="config_file" value="$(find tag_noise_experiment)/param/flir_land_calibration_1440_1080.yaml"/>
    <arg name="start_nodelet_manager" value="false"/>
    <arg name="color" value="false"/>
  </include>

  <node pkg="minimal_bluerov_control_experiment" output="screen" respawn="false" type="binary_PID_experiment.py" name="binary_PID_experiment_node" required="true"> 
    <param name="dry_run" type="bool" value="false"/>
    <rosparam>
      notes: "Single motor testing"
    </rosparam>
  </node>

  <node pkg="rosbag" type="record" name="experiment_rosbag" args="-o /home/bluerov/catkin-workspace/install/share/minimal_bluerov_control_experiment/static_assets/ /camera_array/cam0/camera_info /rosout /bluerov_controller/ar_tag_detector /camera_array/cam0/image_raw/compressed /camera_array/cam0/camera_info /tf /mavros/rc/override /mavros/imu/data_raw">
  </node>
</launch>
